src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.PatternLocalRewriter.cs

```diff
@@ -512,6 +512,25 @@ bool usesOriginalInput(BoundDecisionDagNode node)
 
                 return rewrittenDag;
 
+                static BoundExpression getExpression(int index, BoundObjectCreationExpression expression)
+                {
+                    while (index >= NamedTypeSymbol.ValueTupleRestIndex)
+                    {
+                        if (expression.Arguments[NamedTypeSymbol.ValueTupleRestIndex] is BoundObjectCreationExpression restExpression)
+                        {
+                            expression = restExpression;
+                            index -= NamedTypeSymbol.ValueTupleRestIndex;
+                        }
+                        else
+                        {
+                            // ValueTupleRestIndex should always be another tuple. Hence, it should always have BoundObjectCreationExpression.
+                            Debug.Assert(false, "This should be unreachable.");
```

Comment:
Is my assumption here correct?

Reply:
I believe so. Consider `throw ExceptionUtilities.Unreachable` instead of an assert.

---
In reply to: [492204839](https://github.com/dotnet/roslyn/pull/47893#discussion_r492204839) [](ancestors = 492204839)